### PR TITLE
SMAR-16/ Ajax controller error handling improvements

### PR DESCRIPTION
**Pull request by GytisZum**
SMAR-16/ Ajax controller error handling improvements

Before some of the errors was not handled correctly and there was no fall back scenarios. Also to get a message api request was done all the times. Did some validation to be done on back end without api call

### CODE REVIEW

**Review by tomas862 — COMMENTED** [review]
you could probably use laravel validator, contact Tomas Jasulaitis, he will point to the modules he already uses it. See how it works (although for custom non laravel projects it looks a bit different https://laravel.com/docs/10.x/validation) thats why contact TJ for assistance if you want to use it and if you feel request validation will be more mainstream in this project. 

Also be aware about translations

File: controllers/admin/AdminSmartsuppAjax.php [hunk 1]
```diff
@@ -2,94 +2,144 @@
...
+                    $this->handleLoginAction();
+                    break;
+                case self::CREATE_ACTION:
+                    $this->handleCreateAction();
+                    break;
+                case self::DEACTIVATE_ACTION:
+                    $this->handleDeactivateAction();
+                    break;
+                default:
+                    throw new Exception('Invalid action');
```
Comment (zuk3975) [inline controllers/admin/AdminSmartsuppAjax.php:61]: throwini,o po dvieju eilucii catchini. Nenori tiesiog iskart cia handleError iskviest ir necatchint?

Comment (zuk3975) [reply controllers/admin/AdminSmartsuppAjax.php:61]: ai nebent cia ir kitu methodu exceptionus tikiesi pahandlint tai gal tada ok

File: src/Validator/UserCredentialsValidator.php [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+     * @return void
+     */
+    public function validate($data)
+    {
+        $email = isset($data['email']) ? $data['email'] : '';
+        $password = isset($data['password']) ? $data['password'] : '';
+
+        if (empty($email) || empty($password)) {
+                $this->error = $this->module->l('Empty values provided', self::FILE_NAME);
+                $this->message = $this->module->l('Email and password fields can not be empty', self::FILE_NAME);
```
Comment (zuk3975) [inline src/Validator/UserCredentialsValidator.php:60]: "cannot" dazniau naudojamas

File: src/Validator/UserCredentialsValidator.php [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        if (!$this->validateEmail($email)) {
+            $this->error = $this->module->l('Invalid email format', self::FILE_NAME);
+            $this->message =  $this->module->l('Invalid email address', self::FILE_NAME);
+
+            return;
+        }
+
+        // Validate password
+        if (!$this->validatePassword($password)) {
+            $this->error = $this->module->l('Password length is incorrect', self::FILE_NAME);
```
Comment (zuk3975) [inline src/Validator/UserCredentialsValidator.php]: smulkmena, bet 'invalid' vietoj incorrect butu geriau naudot

File: src/Validator/UserCredentialsValidator.php [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+    /**
+     * @param string $password
+     * @return bool
+     */
+    private function validatePassword($password)
+    {
+        return strlen($password) >= 6 && strlen($password) <= 255;
+    }
+}
```
Comment (zuk3975) [inline src/Validator/UserCredentialsValidator.php]: new line at the end of file missing

File: src/Validator/UserCredentialsValidator.php [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+ * Text Domain:       smartsupp
+ * License:           GPL-2.0+
+ * License URI:       http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+namespace Smartsupp\LiveChat\Validator;
+
+class UserCredentialsValidator
+{
+    const FILE_NAME = 'UserCredentialsValidator';
```
Comment (zuk3975) [inline src/Validator/UserCredentialsValidator.php:26]: netinka klases name perduot ar kazkaip? biski be rysio atrodo

**Review by zuk3975 — APPROVED** [review]
pora niuansu bet siaip ok turbut. Return typu nei vieno nematau, nepalaiko php7.2?

**Comment by GytisZum** [comment]
> you could probably use laravel validator, contact Tomas Jasulaitis, he will point to the modules he already uses it. See how it works (although for custom non laravel projects it looks a bit different https://laravel.com/docs/10.x/validation) thats why contact TJ for assistance if you want to use it and if you feel request validation will be more mainstream in this project.
> 
> Also be aware about translations

This library is very good for validation and I would recommend anyone to use it, unfortunately, but module is compatible with PS 1.6 where symfony http request component is not presented.

**Comment by GytisZum** [comment]
> pora niuansu bet siaip ok turbut. Return typu nei vieno nematau, nepalaiko php7.2?

nuo 5.6

File: controllers/admin/AdminSmartsuppAjax.php (hunk 1 above)
Comment (GytisZum) [reply controllers/admin/AdminSmartsuppAjax.php:61]: Yra libo exceptionai dėl php extensions etc.

File: src/Validator/UserCredentialsValidator.php (hunk 5 above)
Comment (GytisZum) [reply src/Validator/UserCredentialsValidator.php:26]: n moduliu taip padaryta, man ir be ryšio, bet savo funkcija atlieka neblogai

**Comment by tomas862** [comment]
> > you could probably use laravel validator, contact Tomas Jasulaitis, he will point to the modules he already uses it. See how it works (although for custom non laravel projects it looks a bit different https://laravel.com/docs/10.x/validation) thats why contact TJ for assistance if you want to use it and if you feel request validation will be more mainstream in this project.
> > Also be aware about translations
> 
> This library is very good for validation and I would recommend anyone to use it, unfortunately, but module is compatible with PS 1.6 where symfony http request component is not presented.

request is not mandatory for that lib to use as standalone fyi

**Pull request merged**